Dockerfile

```diff
@@ -19,32 +19,50 @@
 # or DOCKER_BUILDKIT=1 docker build -o /tmp/out . --target=buster
 # Generated tarball(s) will end up in /tmp/out
 
+
+# Manually prepare a new enough version of CMake.
+# This should be used on platforms where the default package manager
+# does not provide a new enough version (we require >= 3.12).
+FROM alpine:latest AS cmake-amd64
+
+ENV hash=4d98de8d605da676e71a889dd94f80c76abb377fade2f21e3510e62ece1e1ada
+ADD https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-x86_64.sh \
+    /cmake.sh
+
+FROM alpine:latest AS cmake-arm64
+
+ENV hash=73a35cab2174a3eb8f35083d55c80871185dc3808f3dae3558cd5fbdb29a4614
+ADD https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-aarch64.sh \
+    /cmake.sh
+
+FROM cmake-${BUILDARCH} AS cmake-install
+RUN set -xe; (echo "$hash  /cmake.sh" | sha256sum -c)
+
+
 # ======================================
 # Build Ops Agent for centos-7 
 # ======================================
 
 FROM centos:7 AS centos7-build-base
 
+COPY --from=cmake-install /cmake.sh /cmake.sh
+
 RUN set -x; yum -y update && \
 		yum -y install git systemd \
 		autoconf libtool libcurl-devel libtool-ltdl-devel openssl-devel yajl-devel \
 		gcc gcc-c++ make bison flex file systemd-devel zlib-devel gtest-devel rpm-build java-11-openjdk-devel \
-		expect rpm-sign zip wget && \
+		expect rpm-sign zip && \
 		yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && \
-		yum install -y cmake3 && \
-		ln -fs cmake3 /usr/bin/cmake && \
-		# Download and install CMake manually (zypper only gives us CMake 3.5, but we need >= 3.12)
```

Comment:
Might be worth keeping at least part of this comment (well, with `s/zypper/yum/` or something)…

Reply:
Added a catchall comment to `installCMake`. LMK if that fits or if you'd rather see inline comments so they show up in the Dockerfile.

Also, WDYT about renaming the layer to `cmake-3.25.2-install` so that the version we're actually installing is clear from within compile.go?

Reply:
Missed this one. I think adding the version to the rule would result in a larger change if we ever decide to update it (plus, from what I understand, we don't actually care that the version is 3.25.2, as long as it's recent enough). You *could* rename the rules `cmake-*-recent` (e.g., `cmake-install-recent`) and make the version a build argument (defaulting to 3.25.2)… Alternatively, rename the script to `/cmake-recent.sh` (though I'd still pull out the version into a build argument).

Reply:
Renamed to `cmake-*-recent`.